File: Day035_2025/Q035_Leetcode.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        // Q. Maximum Ascending Subarray Sum
        int maximumAscendingSum = nums[0];
        int currentAscendingSum = nums[0];
        for(int index = 1; index < nums.size(); ++index){
            if(nums[index] > nums[index - 1]){
                currentAscendingSum += nums[index];
            }
            else{
                maximumAscendingSum = max(maximumAscendingSum, currentAscendingSum);
                currentAscendingSum = nums[index];
            }
        }
        maximumAscendingSum = max(maximumAscendingSum, currentAscendingSum);

        return maximumAscendingSum;
    }
};

File: Day035_2025/Q035_Leetcode.py
from typing import List

class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        maximum_ascending_sum = nums[0]
        current_ascending_sum = nums[0]

        for index in range(1, len(nums)):
            if nums[index] > nums[index - 1]:
                current_ascending_sum += nums[index]
            else:
                maximum_ascending_sum = max(maximum_ascending_sum, current_ascending_sum)
                current_ascending_sum = nums[index]

        return max(maximum_ascending_sum, current_ascending_sum)
